eclair-core/src/main/scala/fr/acinq/eclair/db/Databases.scala

```diff
@@ -162,6 +174,71 @@ object Databases extends Logging {
         }
       }
 
+      safetyChecks_opt foreach { initChecks =>
+
+        PgUtils.inTransaction { connection =>
+          using(connection.createStatement()) { statement =>
+
+            def checkMaxAge(name: String, maxAge: FiniteDuration, sqlQuery: String): Unit = {
+              import ExtendedResultSet._
+              val smallestAge_opt = statement
+                .executeQuery(sqlQuery)
+                .headOption // sql max() will always return a result, with perhaps a null value if there was no records
+                .flatMap(_.getTimestampNullable("max"))
+                .map(ts => TimestampMilli.now() - TimestampMilli.fromSqlTimestamp(ts))
+              require(smallestAge_opt.isDefined, s"db check failed: no $name found")
+              require(smallestAge_opt.get <= maxAge, s"db check failed: most recent $name is too old (${smallestAge_opt.get.toMinutes} minutes > ${maxAge.toMinutes} minutes)")
+              logger.info(s"db check ok: max age ${smallestAge_opt.get.toMinutes} minutes <= ${maxAge.toMinutes} minutes for $name")
+            }
+
+            checkMaxAge(name = "local channel",
```

Comment:
This check will always fail for a node that has no channels.
Would it make sense to skip it if we don't have any channels or if they are all closed?
Otherwise node operators will need to first disable the checks in their config, open channels, then re-enable it.

I think it's fine to let this check fail when eclair has no channel, but then we should probably mention it in `reference.conf` and tell node operators to only enable the checks once they expect their node to be a somewhat busy node.

Reply:
> This check will always fail for a node that has no channels.

Yes, that's why it's disabled by default.

> I think it's fine to let this check fail when eclair has no channel, but then we should probably mention it in reference.conf and tell node operators to only enable the checks once they expect their node to be a somewhat busy node.

This check only makes sens for an already established node, because the values are entirely dependent on how big and how busy the node is. To be more precise, routing table-related values are independent, but they are much less important. What matters is local channels. This was missing from the doc, I've added a paragraph in cd1905115 to hopefully make it more clear.